src/role-access/containers/exclusion/check-answers/check-answers.component.ts

```diff
@@ -1,31 +1,64 @@
-import { Store } from '@ngrx/store';
-import { ExclusionNavigationEvent, ExclusionState } from '../../../models';
+import { Component, Input, OnDestroy } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { Subscription } from 'rxjs';
+import { Answer, ExcludeOption, ExclusionNavigationEvent, ExclusionState, ExclusionStateData } from '../../../models';
 import { ExclusionNavigation } from '../../../models/exclusion-navigation.interface';
 import * as fromFeature from '../../../store';
+import { AnswerHeaderText, AnswerLabelText } from '../../../models/enums'
 
 @Component({
   selector: 'exui-check-answers',
-  templateUrl: './check-answers.component.html',
-  styleUrls: ['./check-answers.component.scss']
+  templateUrl: './check-answers.component.html'
 })
-export class CheckAnswersComponent implements OnInit {
+export class CheckAnswersComponent implements OnDestroy {
 
   @Input() public navEvent: ExclusionNavigation;
 
+  public answers: Answer[] = [];
+  public caption = AnswerHeaderText.AddExclusion;
+  public heading = AnswerHeaderText.CheckAnswers;
+  public hint = AnswerHeaderText.CheckInformation;
+  public storeSubscription: Subscription;
+
+  private exclusionStateData: ExclusionStateData;
+
   constructor(private readonly store: Store<fromFeature.State>) {
+    this.storeSubscription = this.store.pipe(select(fromFeature.getRoleAccessState)).subscribe(exclusion => this.setAnswersFromExclusionStore(exclusion));
   }
 
-  public ngOnInit(): void {
+  public ngOnDestroy() {
+    if (this.storeSubscription) {
+      this.storeSubscription.unsubscribe();
+    }
   }
 
   public navigationHandler(navEvent: ExclusionNavigationEvent) {
     switch (navEvent) {
-      case ExclusionNavigationEvent.CONTINUE:
-        this.store.dispatch(new fromFeature.ChangeNavigation(ExclusionState.FIND_PERSON));
+      case ExclusionNavigationEvent.CONFIRM_EXCLUSION:
+        this.store.dispatch(new fromFeature.ConfirmExclusionAction(this.exclusionStateData));
         break;
       default:
         throw new Error('Invalid option');
     }
   }
+
+  public onNavigate(action) {
+    this.store.dispatch(new fromFeature.ChangeNavigation(action));
+  }
+
+  private setAnswersFromExclusionStore(exclusion: ExclusionStateData): void {
+    this.exclusionStateData = exclusion;
+    this.answers = [];
+    this.answers.push({ label: AnswerLabelText.WhoFor, value: exclusion.exclusionOption, action: ExclusionState.CHOOSE_EXCLUSION });
+    if (exclusion.exclusionOption === ExcludeOption.EXCLUDE_ANOTHER_PERSON) {
+      this.answers.push({ label: AnswerLabelText.PersonRole, value: exclusion.personRole, action: ExclusionState.CHOOSE_PERSON_ROLE });
+      let personDetails = exclusion.person.name;
+      if (exclusion.person.email) {
+        personDetails += '\n' + exclusion.person.email;
+      }
+      this.answers.push({ label: AnswerLabelText.Person, value: personDetails, action: ExclusionState.FIND_PERSON });
+    };
```

Comment:
redundant ;

Reply:
I've removed it now - will be available on next commit.